tests/specs/mcd/verification.k

```diff
@@ -19,27 +19,54 @@ module VERIFICATION-SYNTAX
-    // rule #WordPackAddrUInt8        (     ADDR , UINT8               ) => UINT8 *Int pow160 +Int ADDR                              requires #rangeAddress(ADDR) andBool #rangeUInt(8, UINT_8)
+    rule    ADDR_UINT48_UINT48 ==K #WordPackAddrUInt48UInt48(ADDR, UINT48_1, UINT48_2)
+         => ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48
+    andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+    andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+    andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+
+    rule    UINT48_UINT48 ==K #WordPackUInt48UInt48(UINT48_1, UINT48_2)
+         => UINT48_1 ==Int maxUInt48 &Int  UINT48_UINT48
+    andBool UINT48_2 ==Int maxUInt48 &Int (UINT48_UINT48 /Int pow48)
+    andBool #rangeUInt(96, UINT48_UINT48)
+
+    rule    ADDR_UINT8 ==K #WordPackAddrUInt8(ADDR, UINT8)
+         => ADDR  ==Int maxUInt160 &Int  ADDR_UINT8
+    andBool UINT8 ==Int maxUInt8   &Int (ADDR_UINT8 /Int pow160)
+    andBool #rangeUInt(168, ADDR_UINT8)
+
+    //
+    // When updating variables which originally contained `WordPack*`, the Solidity compiler will mask the original value everywhere _except_ where you're updating.
+    // Then the update will be bitwise `|Int` with the masked value, and the new value written back.
+    // Here we provide the masks which are used by the Solidity compiler, and rules which fold these bitwise updates back up into semantic `#WordPack*` arguments for the purposes of matching the RHS of proofs.
+    //
+
+    syntax Int ::= "maskWordPackUInt48UInt48_1" | "maskWordPackUInt48UInt48_2"
+ // --------------------------------------------------------------------------
+    rule maskWordPackUInt48UInt48_1 => 115792089237316195423570985008687907853269984665561335876943319951794562400255 [macro] // pow256 -Int pow96 +Int pow48 -Int 1
+    rule maskWordPackUInt48UInt48_2 => 115792089237316195423570985008687907853269984665640564039457583726438152929280 [macro] // pow256 -Int pow48
 
-    rule maxUInt48 &Int  UINT48_UINT48             => UINT48_1 requires UINT48_UINT48 ==Int #WordPackUInt48UInt48(UINT48_1, UINT48_2) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
-    rule maxUInt48 &Int (UINT48_UINT48 /Int pow48) => UINT48_2 requires UINT48_UINT48 ==Int #WordPackUInt48UInt48(UINT48_1, UINT48_2) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
+    rule  UINT48_1             |Int (maskWordPackUInt48UInt48_2 &Int UINT48_UINT48) => #WordPackUInt48UInt48( UINT48_1 , (maskWordPackUInt48UInt48_2 &Int UINT48_UINT48) /Int pow48 ) requires #rangeUInt(256, UINT48_UINT48) andBool #rangeUInt(48, UINT48_1)
+    rule (UINT48_2 *Int pow48) |Int (maxUInt48                  &Int UINT48_UINT48) => #WordPackUInt48UInt48( maxUInt48 &Int UINT48_UINT48 , UINT48_2                               ) requires #rangeUInt(256, UINT48_UINT48) andBool #rangeUInt(48, UINT48_2)
```

Comment:
Isn't it `#rangeUInt(96, UINT48_UINT48)`?

The second rule might be fine even if the higher bits are non-zero, but the first rule would _not_.  The higher bits are not cleared by ANDing with `maskWordPackUInt48UInt48_2`.

Reply:
BTW, these rules are quite confusing mainly because of the reverse order of the arguments being packed.  Would be the spec hard to read if we reverse the order of the arguments in the WordPack* terms?

Reply:
Good catch, I've changed it to `#rangeUInt(96, UINT48_UINT48)`.

I would prefer not to change the argument order for the WordPack, because that's the order the KLab currently is generating them in. It's not huge change to make for it, but would rather focus on getting more proofs passing.

If you really want me to change the argument order (instead of just leaving the comment about it), let me know.